sig-network/link-state-proposal.md

```diff
@@ -0,0 +1,211 @@
+
+When using the functionality purposed by this enhancement with network binding plugins using a sidecar container -
+it is up to the plugin's vendor to support this functionality - as they control the interface section in the domain XML.
+
+### Link State Reporting
+
+A new field will be added to the VMI interface status that will be used for reporting the interface's current link state (up/down):
+```go
+type VirtualMachineInstanceNetworkInterface struct {
+	...
+	LinkState string `json:"linkState,omitempty"`
+	...
+}
+```
+
+virt-handler will be responsible for reporting this field's value, based on domain spec information received from virt-launcher.
+An empty specification (as exists today) would be considered as `up`.
+
+> [!NOTE]
+> The linkState information will be reported for interfaces which are present in the VirtualMachineInstance's spec.
+> Interfaces defined inside the guest will not have their link state reported.
+>
+> Changing the link state of an interface inside the guest or in the underlying network provider - will not affect this report.
+
+## API Examples
+### Controlling The Link State
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: my-vm
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+            - name: default
+              state: down
+              masquerade: { }
+      networks:
+        - name: default
+          pod: { }
+```
+### Link State Reporting
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: my-vm
+spec:
+  domain:
+    devices:
+      interfaces:
+      - name: default
+        state: down
+        masquerade: { }
+  networks:
+  - name: default
+    pod: { }
+status:
+  interfaces:
+    - name: default
+      linkState: down
+```
+
+## Alternatives
+
+## Scalability
+
+This enhancement does not affect scalability.
+
+## Update/Rollback Compatibility
+
+As this functionality will require the cooperation of the virt-launcher component, it could only be supported by
+up-to-date virt-launcher pods.
+
+## Functional Testing Approach
+
+The following e2e scenarios will be added:
+- A VM started with interface with a link state set to `down`, then changed to `up`.
+- Setting a link state of a running VM to `down`, then `up`.
+- Hot-plugging an interface connected to a secondary network with link state set to `down`, then changed to `up`.
+- Migrating a VM that has an interface with a link state set to `down`, then changed to `up`.
+
+## Implementation Phases
+
+1. Addition of the new `InterfaceState` values.
+2. Adjusting the network validator to accept the new `InterfaceState` values.
+3. Adjusting virt-launcher:
+   - On creation
+   - When VM is running
+   - On hotplug
+4. Addition of e2e tests.
+5. Addition of the `LinkState` field to `VirtualMachineInstanceNetworkInterface`.
+6. Adjusting virt-handler's reporting logic.
+7. Adjusting e2e tests.
+
+## Feature lifecycle Phases
```

Comment:
The feature lifecycle document [1] suggests that we always start with a non-GA FG and then graduate. 

While I can see an argument to skip this requirement entirely for this feature and just GA it straight away I wonder if we should instead compromise with a beta FG for the v1.5.0 release we can then GA in v1.6.0?

@EdDev @vladikr @xpivarc I'd appreciate your thoughts here.

IMHO we should also extend the template and/or lifecycle document [1] with the possibility of starting with a beta FG if folks agree to it here.

[1] https://github.com/kubevirt/community/blob/main/design-proposals/feature-lifecycle.md

Reply:
I am in favor to be strict with regards of new features, however, in case a proper reasoning can be provided, we can enforce an exception on the FG path.
I think this one is a fit to bypass any FG, it is too trivial to require an evaluation/experiment stage.

I asked to give a more detailed convincing explanation on why GA first is a good fit [1].

I vote to directly approve this small feature for GA.

[1] https://github.com/kubevirt/enhancements/pull/2#discussion_r1918986205

Reply:
I've added a few points, following this comment and @EdDev's [comment](https://github.com/kubevirt/enhancements/pull/2#discussion_r1918986205).